infrastructure/src/main/java/org/corfudb/infrastructure/log/BoundedMap.java

```diff
@@ -0,0 +1,116 @@
+package org.corfudb.infrastructure.log;
+
+import com.google.common.base.Preconditions;
+
+import java.util.Iterator;
+import java.util.concurrent.atomic.AtomicLongArray;
+
+public class BoundedMap {
+
+    private final long offset;
+    private final AtomicLongArray array;
+    public static final long NOT_SET = -1;
+
+    public BoundedMap(long offset, int size) {
+        Preconditions.checkArgument(offset >= 0);
+        Preconditions.checkArgument(size > 0);
+        Preconditions.checkArgument(Math.addExact(offset, size) < Integer.MAX_VALUE);
+        this.offset = offset;
+        this.array = new AtomicLongArray(size);
+        for (int idx = 0; idx < array.length(); idx++) {
+            this.array.set(idx, NOT_SET);
+        }
+    }
+
+    private void checkRange(long num) {
+        if (num < offset || num >= offset + array.length()) {
+            throw new IllegalArgumentException(num + " not in [" + offset + ", " + (offset + array.length()) + ")");
+        }
+    }
+
+    private long pack(long high, int low) {
+        return (((long) high) << 28) | (low & 0x000000000fffffffL); //0xfffffffff0000000L
```

Comment:
This mask has to be the same as the Segment.MAX_SEGMENT_SIZE. Can we derive the mask from there instead of hardcoding it here?

Reply:
I'll remove this. Since pack has been moved `Segment::Index`.